packages/loader/container-loader/src/naming.md

```diff
@@ -1,28 +1,27 @@
 * `IContainer` (public):
   * `connectionState: ConnectionState`
     * `Disconnected`: no socket connection to delta server
-    * `Connecting`: socket connection established but we're not caught up
+    * `Pending`: socket connection established but we're not caught up
       * Unable to send outbound messages
       * Maybe inbound is still paused too, depends on `IContainerLoadMode.opsBeforeReturn`
     * `Connected`: connected and caught up
       * Not quite true for Read connection, see PR #9377
       * ClientId may not yet be in audience, see Issue #7275
       * Inbound/Outbound are operational (...is there a possible race condition if `IContainerLoadMode.opsBeforeReturn === undefined`?)
-  * `connected: boolean`
-    * `connectionState === ConnectionState.Connected`
-  * `resume()`
-    * Calls `DeltaManager.connect`
-  * `setAutoReconnect(reconnect: boolean)`
-    * If false, will disconnect from delta stream.  If true, will probably connect to delta stream
+  * `connect()`
+    * Calls `DeltaManager.connect`  (and picks up some semantics from old `setAutoReconnect`?)
+  * `disconnect()`
+    * Disconnect from delta stream, until `connect()` is called again.
   * events emitted:
-    * `connected`: when transitioned to Connected state
-    * `disconnected`: when transitioned to Disconnected state
+    * `connecting`: when starting to establish connection. Some kind of progress tracker passed for observability of estabilishing healthy connection to delta stream
```

Comment:
If we really want to expose this, I'd probably roll it into state machine (i.e. ConnectionState and appropriate events). I.e. modes can be
- disconnected (and not trying to connect, like initial container state or after Container.disconnect() call)
- connecting
- pending
- connected.

This will make it easier for everybody to reason about all the states, including us (see comments in Scott's review, where we might still connect after Container.disconnect() call, because we already have pending connection in flight).

Reply:
Makes sense.  We'll have to wait some time to introduce it (or give it a different name) due to conflict with existing "Connecting" state (which will become "Pending").

After all the time I've spent with FFX trying to diagnose slow connections, I had the idea for the host to receive a progress tracker object to communicate pretty fine-grained progress through the connection flow. Basically tracking each step/prerequisite and if it succeeds/fails:

* Starting connectCore loop
* JoinSession Token Fetch
* (GetWebsocketToken Token Fetch if applicable)
* (GetWebsocketToken if applicable)
* Initiated socket.io

Obviously needs to be fleshed out.  But in any case, maybe the connectionStateChanged event could have an optional 2nd event arg which would be this progress tracker, and maybe it's only passed for `connecting`?